integration-tests/src/test/java/oracle/kubernetes/operator/ITServerDiscovery.java

```diff
@@ -0,0 +1,541 @@
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * <p>It also create operator and verify its deployed successfully. Create domain and verify
+   * domain is created.
+   *
+   * @throws Exception
+   */
+  @BeforeClass
+  public static void staticPrepare() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    // initialize test properties and create the directories
+    initialize(appPropsFile);
+
+    // create operator1
+    if (operator == null) {
+      logger.info("Creating Operator & waiting for the script to complete execution");
+      operator = TestUtils.createOperator(operatorYmlFile);
+    }
+
+    // create domain
+    if (domain == null) {
+      logger.info("Creating WLS Domain & waiting for the script to complete execution");
+      domain = TestUtils.createDomain(domainonpvwlstFile);
+      domain.verifyDomainCreated();
+    }
+
+    domainYaml =
+        BaseTest.getUserProjectsDir()
+            + "/weblogic-domains/"
+            + domain.getDomainUid()
+            + "/domain.yaml";
+
+    // create test domain yaml file
+    testDomainYamlFile = BaseTest.getResultDir() + "/" + testYamlFileName;
+
+    Path sourceFile = Paths.get(domainYaml);
+    Path targetFile = Paths.get(testDomainYamlFile);
+
+    Files.copy(sourceFile, targetFile, StandardCopyOption.REPLACE_EXISTING);
+
+    String content = new String(Files.readAllBytes(targetFile), StandardCharsets.UTF_8);
+    content = content.replaceAll("replicas: 2", "replicas: 3");
+    Files.write(targetFile, content.getBytes(StandardCharsets.UTF_8));
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories
+   *
+   * @throws Exception
+   */
+  @AfterClass
+  public static void staticUnPrepare() throws Exception {
+    logger.info("+++++++++++++++++++++++++++++++++---------------------------------+");
+    logger.info("BEGIN");
+    logger.info("Run once, release cluster lease");
+
+    StringBuffer cmd =
+        new StringBuffer("export RESULT_ROOT=$RESULT_ROOT && export PV_ROOT=$PV_ROOT && ");
+    cmd.append(BaseTest.getProjectRoot())
+        .append("/integration-tests/src/test/resources/statedump.sh");
+    logger.info("Running " + cmd);
+
+    ExecResult result = ExecCommand.exec(cmd.toString());
+    if (result.exitValue() == 0) logger.info("Executed statedump.sh " + result.stdout());
+    else
+      logger.info("Execution of statedump.sh failed, " + result.stderr() + "\n" + result.stdout());
+
+    if (domain != null && !SMOKETEST) domain.shutdownUsingServerStartPolicy();
+
+    if (JENKINS) {
+      cleanup();
+    }
+
+    if (getLeaseId() != "") {
+      logger.info("Release the k8s cluster lease");
+      TestUtils.releaseLease(getProjectRoot(), getLeaseId());
+    }
+
+    logger.info("SUCCESS");
+  }
+
+  /**
+   * Verify that a running Operator connects to a pre-configed and newly started MS
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testOPConnToNewMS() throws Exception {
```

Comment:
Add Assume.assumeFalse(QUICKTEST) at the beginning of all tests as we don't want to run these tests on Wercker.

Reply:
Vanaja, I doubt we should do it.  Please check the changes in "Stop running server discovery tests on Wercker", I had added Assume.assumeFalse(QUICKTEST) in tests, but run failed on Wercker. Per this link:
https://stackoverflow.com/questions/1689242/conditionally-ignoring-tests-in-junit-4
it says:

 If you do it in the test itself, your @Before method will get run. 

That's why I submmited another changes afterward, "refactor server discovery tests to stop running it on Wercker" to ONLY add it in statisPrepare

Reply:
Yes, you are right. If you do for @BeforeClass we are good. no need to do for every test. But should be done in @AfterClass as well.

Reply:
agree. Will do

Reply:
No need to do it in @afterclass